lib/clerk/utils.rb

```diff
@@ -1,11 +1,46 @@
+require 'base64'
+
 module Clerk
   module Utils
+    PUBLISHABLE_KEY_LIVE_PREFIX = 'pk_live_'.freeze
+    PUBLISHABLE_KEY_TEST_PREFIX = 'pk_test_'.freeze
+
     module_function
     def camelize(term)
       string = term.to_s
       string = string.sub(/^[a-z\d]*/) { match.capitalize }
       string.gsub!(/(?:_|(\/))([a-z\d]*)/) { "#{$1}#{$2.capitalize}" }
       string
     end
+
+    # see: https://github.com/clerkinc/javascript/blob/main/packages/backend/src/util/parsePublishableKey.ts
+    module_function
+    def parse_publishable_key(key)
+      return unless publishable_key?(key)
+
+      frontend_api = decode_publishable_key(key)
+      return unless frontend_api&.end_with?('$')
+
+      { instance_type: instance_type(key), frontend_api: frontend_api.chop }
+    end
+
+    module_function
+    def publishable_key?(key)
+      valid_prefix = key&.start_with?(PUBLISHABLE_KEY_LIVE_PREFIX) || key.start_with?(PUBLISHABLE_KEY_TEST_PREFIX);
+      valid_fapi_postfix = decode_publishable_key(key)&.end_with?('$')
+
+      valid_prefix && valid_fapi_postfix
+    end
+
+    module_function
+    def instance_type(key)
+      key&.start_with?(PUBLISHABLE_KEY_LIVE_PREFIX) ? 'production' : 'development';
+    end
+
+    module_function
+    def decode_publishable_key(key)
+      encoded_part = key.split('_')[2] if key
+      Base64.decode64(encoded_part) if encoded_part
+    end
   end
```

Comment:
:upside_down_face: These methods are good candidates for a `PublishableKey` class. Something like:

```ruby
class PublishableKey
  attr_reader :frontend_api, :environment_type
  def self.parse(key)
    @key = key
  end
  
  def decode
  end
  
  def valid?
  end
end
```

Reply:
i like this one. i will update the code.
Looking at clerk rails starter i will we should provide a way to expose the cleckJS url since currently the developers will have to add the logic of parsing the publishable key and using the frontend_api to determine the clerkJS url (see application_helper in clerk rails starter)
ref: https://github.com/clerkinc/clerk-rails-starter/pull/4/files

Reply:
After discussing it offline, i have removed the `Clerk::PublishableKey` since it's not directly related to this PR and i will open a discussion in slack to discuss how we should proceed with handling the parsing of publishable keys in our backend SDKs.